### PR TITLE
Cert-manager

### LINKED ISSUE #72: Cert Manager
See if we can get certificates auto issued to services

### PR DESCRIPTION
**Pull request by njbarber**
This PR closes #72 by adding cert-manager to manage certificates. It removes the existing certificates task and creates cluster issuer resources to create secrets in namespaces where certificate resources are present.

### CODE REVIEW

**Comment by aperullo** [comment]
In `grayskull.yml` the certs role task is still present. Also perhaps the templating of the CSR should happen in the certs-manager role if you intend on also making the CA in there.

File: kubernetes/ansible/playbooks/roles/cert_manager_role/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+task_modes:
+  - download
+  - install
+
+bin_dir: "/usr/local/bin"
```
Comment (aperullo) [inline kubernetes/ansible/playbooks/roles/cert_manager_role/defaults/main.yml]: `bin_dir` should already be available as a variable from higher up in the playbooks.

File: kubernetes/ansible/playbooks/roles/cert_manager_role/templates/cert-manager-namespace.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ cert_manager_namespace }}
+  labels:
+    name: {{ cert_manager_namespace }}
+    certmanager.k8s.io/disable-validation: "true"
```
Comment (aperullo) [inline kubernetes/ansible/playbooks/roles/cert_manager_role/templates/cert-manager-namespace.yml]: Whats this for?

Comment (njbarber) [reply kubernetes/ansible/playbooks/roles/cert_manager_role/templates/cert-manager-namespace.yml]: It's necessary when deploying the custom resource definitions. See:

https://github.com/helm/charts/tree/master/stable/cert-manager#installing-the-chart

File: kubernetes/ansible/playbooks/roles/cert_manager_role/defaults/main.yml (hunk 1 above)
Comment (njbarber) [reply kubernetes/ansible/playbooks/roles/cert_manager_role/defaults/main.yml]: Yep. Related, I define `grayskull_ca_issuer_name` and `kube_ca_issuer_name` in the defaults for cert-manager, and define them again under defaults for any role that needs to use. Should these be made higher-level variables as well?

File: kubernetes/ansible/playbooks/roles/cert_manager_role/tasks/install.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+- name: Cert-manager | Check for cfssljson
+  stat:
+    path: "{{ bin_dir }}/cfssljson"
+  register: cfssljson_exists
+
+- name: Cert-manager | Install cfssl
+  block:
+    - name: Cert-manager | Download cfssl binary
+      get_url:
+        url: https://pkg.cfssl.org/R1.2/cfssl_linux-amd64
```
Comment (aperullo) [inline kubernetes/ansible/playbooks/roles/cert_manager_role/tasks/install.yml]: Should probably move these out to variables.

File: kubernetes/ansible/playbooks/roles/cert_manager_role/tasks/install.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        dest: "{{ bin_dir }}/cfssl"
+        mode: '0755'
+
+    - name: Cert-manager | Download cfssljson binary
+      get_url:
+        url: https://pkg.cfssl.org/R1.2/cfssljson_linux-amd64
```
Comment (aperullo) [inline kubernetes/ansible/playbooks/roles/cert_manager_role/tasks/install.yml]: Should probably move these out to variables.

File: kubernetes/ansible/playbooks/roles/cert_manager_role/defaults/main.yml (hunk 1 above)
Comment (aperullo) [reply kubernetes/ansible/playbooks/roles/cert_manager_role/defaults/main.yml]: Yeah, since they are used everywhere I think thats reasonable

Comment (aperullo) [reply kubernetes/ansible/playbooks/roles/cert_manager_role/defaults/main.yml]: Yeah, thats reasonable

File: kubernetes/ansible/playbooks/roles/cert_manager_role/tasks/install.yml (hunk 3 above)
Comment (njbarber) [reply kubernetes/ansible/playbooks/roles/cert_manager_role/tasks/install.yml]: Done.

File: kubernetes/ansible/playbooks/roles/cert_manager_role/tasks/install.yml (hunk 4 above)
Comment (njbarber) [reply kubernetes/ansible/playbooks/roles/cert_manager_role/tasks/install.yml]: Done.

**Pull request merged**